internal/handlers/pg/pgdb/indexes.go

```diff
@@ -55,43 +56,79 @@ func Indexes(ctx context.Context, tx pgx.Tx, db, collection string) ([]Index, er
 		res[i] = idx.Index
 	}
 
-	// TODO Add tests that indexes sorted correctly: https://github.com/FerretDB/FerretDB/issues/1509
-	slices.SortFunc(res, func(a, b Index) bool { return a.Name < b.Name })
-
 	return res, nil
 }
 
-// createIndex creates a new index for the given params.
-// TODO This method will become exported in https://github.com/FerretDB/FerretDB/issues/1509.
-func createIndex(ctx context.Context, tx pgx.Tx, db, collection string, i *Index) error {
+// CreateIndexIfNotExists creates a new index for the given params if such an index doesn't exist.
+//
+// If the index exists, it doesn't return an error.
+// If the collection doesn't exist, it will be created and then the index will be created.
+func CreateIndexIfNotExists(ctx context.Context, tx pgx.Tx, db, collection string, i *Index) error {
+	if err := CreateCollectionIfNotExists(ctx, tx, db, collection); err != nil {
+		return err
+	}
+
 	pgTable, pgIndex, err := newMetadataStorage(tx, db, collection).setIndex(ctx, i.Name, i.Key, i.Unique)
 	if err != nil {
 		return err
 	}
 
-	if err := createPgIndexIfNotExists(ctx, tx, db, pgTable, pgIndex, true); err != nil {
+	if err := createPgIndexIfNotExists(ctx, tx, db, pgTable, pgIndex, i.Key, i.Unique); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 // createPgIndexIfNotExists creates a new index for the given params if it does not exist.
-func createPgIndexIfNotExists(ctx context.Context, tx pgx.Tx, schema, table, index string, isUnique bool) error {
+func createPgIndexIfNotExists(ctx context.Context, tx pgx.Tx, schema, table, index string, fields IndexKey, isUnique bool) error {
+	if len(fields) == 0 {
+		return lazyerrors.Errorf("no fields for index")
+	}
```

Comment:
To test and add non-lazy error https://github.com/FerretDB/FerretDB/issues/2311

Reply:
Yes, it is marked as not tested because we handle this case on the handler level itself. I didn't want to "pushdown" all the validation to the pgdb level as in this case we will have to introduce even more new error variables here for each different validation rule that requires a different error code to be returned. If the problem is solved, could you please close the test? (or suggest what needs to be done otherwise)